server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,236 @@
+first_path = sys.path[0]
+assert first_path.endswith('/bin')
+lib_path = os.path.join(os.path.dirname(first_path), 'lib')
+sys.path[0] = lib_path
+
+from pbench import init_report_template, report_status, _rename_tb_link, \
+                       PbenchConfig, BadConfig, get_es, get_pbench_logger, quarantine
+from s3backup import S3Config
+
+_NAME_ = "pbench-backup-tarballs"
+
+# The link source and destination for this operation of this script.
+_linksrc = "TO-BACKUP"
+_linkdest = "BACKED-UP"
+
+# Global logger for the module, setup in main()
+_logger = None
+
+def backup_data(s3_obj, config, report, logger):
+    tarlist = glob.iglob('{}/*/{}/*.tar.xz'.format(config.ARCHIVE, _linksrc))
+    qdir = config.get('pbench-server', 'pbench-quarantine-dir')
+    try:
+        list_bucket_objects = s3_obj.connector.list_objects(
+            Bucket='{}'.format(s3_obj.bucket_name))
+    except Exception as e:
+        logger.info(
+            "Bucket: {} is not present in s3\n{}".format(s3_obj.bucket_name, e))
+        return
+    try:
+        s3_contents = [s3_content_obj['Key']
+                        for s3_content_obj in list_bucket_objects['Contents']]
+    except Exception:
+        s3_contents = []
+
+    nbackup, ns3_success, ns3_fail, nquaran = 0, 0, 0, 0
+    for tb in sorted(tarlist):
+        tar = os.path.realpath(tb)
+        resultname = os.path.basename(tar)
+        controller_path = os.path.dirname(tar)
+        controller = os.path.basename(controller_path)
+        s3_resultname = "{}/{}".format(controller, resultname)
+        backup_controller_path = "{}/{}".format(config.BACKUP, controller)
+        try:
+            os.mkdir(backup_controller_path)
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.exception(
+                "os.mkdir: Unable to create backup destination directory: {}".format(backup_controller_path))
+            raise
+        backup_contents = os.listdir(backup_controller_path)
+        archive_md5 = ("{}.md5".format(tar))
+        with open(archive_md5) as f:
+            archive_md5_hex_value = f.readline().split(" ")[0]
+
+        # Check if the result already present in backup or not
+        if resultname in backup_contents:
+            backup_md5 = ("{}/{}.md5".format(backup_controller_path, resultname))
+            with open(backup_md5) as f:
+                backup_md5_hex_value = f.readline().split(" ")[0]
+            if archive_md5_hex_value == backup_md5_hex_value:
+                # declare success
+                nbackup += 1
+                logger.info(
+                    "Locally Backed-up Successfully: {}\n".format(resultname))
+            else:
+                # quarantine the result and move the symlink
+                # from _linksrc to _linkdest
+                quarantine(qdir, logger, tar, archive_md5)
+                nquaran += 1
+                _rename_tb_link(tb, os.path.join(
```

Comment:
@portante I had an impression that we quarantine the actual tarball and its md5 because prep-shim and some other script does like that. I had a discussion with @ndokos regarding it. Let finalize what we want to quarantine: the actual tarball or only the symlink. According to it, I will update this part.